src/Perimeter.js

```diff
@@ -0,0 +1,174 @@
+type EventListener = (MouseEvent) => void;
+
+type Props = {
+  onBreach: () => void,
+  once?: boolean,
+  padding: number,
+  mapListeners?: (EventListener) => EventListener,
+  children?: React$Element<*>
+};
+
+export default class Perimeter extends Component {
+  props: Props;
+  node: null | HTMLElement;
+  bounds: null | ClientRect;
+  initialOffset: number;
+  breached: boolean;
+  listening: boolean;
+  constructor(props: Props) {
+    const { mapListeners } = props;
+    super(props);
+    // The HTML element used as the perimeter center
+    this.node = null;
+    // The result of calling getBoundingClientRect on this.node
+    this.bounds = null;
+    // Initial pageYOffset, required since pages may be scrolled on load
+    // and getBoundingClientRect will take that into consideration when
+    // calculating the top offset
+    this.initialOffset = 0;
+    // Whether the mouse is within the perimeter
+    this.breached = false;
+    // If we're still listening for mousemove/resize events
+    this.listening = false;
+    // If a mapListeners function is provided, update the instance event
+    // handlers to be the returned mapped versions. Common use case would be
+    // debounced or throttled listeners.
+    if (mapListeners) {
+      this.handleMouseMove = mapListeners(this.handleMouseMove);
+      this.handleResize = mapListeners(this.handleResize);
+    }
+  }
+
+  static propTypes = {
+    onBreach: PropTypes.func.isRequired,
+    once: PropTypes.bool,
+    padding: PropTypes.number.isRequired,
+    mapListeners: PropTypes.func
+  };
+
+  static contextTypes = {
+    mouseMove: PropTypes.shape({
+      subscribe: PropTypes.func.isRequired,
+    })
+  };
+
+  /**
+   * When the component mounts we calculate the ClientRect
+   * for the target node and attach event listeners for:
+   *   - `mousemove` for checking perimeter breaches
+   *   - `resize` for recalculating ClientRect
+   */
+  componentDidMount() {
+    const { handleMouseMove, handleResize, node } = this;
+    if (node) {
+      this.bounds = node.getBoundingClientRect();
+      this.initialOffset = window.pageYOffset;
+      if(this.context && this.context.mouseMove && this.context.mouseMove.subscribe){
+        this.mouseMoveUnsub = this.context.mouseMove.subscribe(handleMouseMove)
+      } else {
+        window.addEventListener('mousemove', handleMouseMove);
+      }
+      window.addEventListener('resize', handleResize);
+      this.listening = true;
+    }
+  }
+
+  /**
+   * Remove event listeners on unmount if we are still
+   * listening.
+   */
+  componentWillUnmount() {
+    if (this.listening) {
+      this.removeEventListeners();
+    }
+  }
+
+  /**
+   * Removes the `mousemove` and `resize` listeners. May
+   * be called on unmount, or after `onBreach` is called
+   * if the `once` prop is set to `true`
+   */
+  removeEventListeners() {
+    if(this.mouseMoveUnsub && this.context && this.context.mouseMove && this.context.mouseMove.subscribe){
```

Comment:
Same here, we can simplify this check with the changes mentioned [here](https://github.com/aweary/react-perimeter/pull/7/files#r106789178) in mind

```js
const { mouseMove } = this.context;
if (mouseMove && mouseMove.unsubscribe) {
   mouseMove.unsubscribe(this.handleMouseMove)
}
```